src/components/SearchBox.js

```diff
@@ -0,0 +1,60 @@
+import React, { useRef, useEffect, useContext, useState } from 'react';
+import './SearchBox.css';
+import { LocationsContext } from '../context/LocationsContext';
+import { Icon } from '@iconify/react';
+import searchIcon from '@iconify-icons/bi/search';
+
+export default function SearchBox() {
+  const { coordinates, mapsApi, mapInstance, setNearByPlaces } = useContext(
+    LocationsContext,
+  );
+
+  const inputRef = useRef(null);
+
+  const [searchTerm, setSearchTerm] = useState('');
+  const [error, setError] = useState(false);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setSearchTerm(inputRef.current.value);
+  };
+
+  useEffect(() => {
+    if (searchTerm && mapsApi) {
+      const callback = (results, status) => {
+        if (status === 'ZERO_RESULTS') {
+          console.log('error');
+          setError(true);
+        }
+        if (status === mapsApi.places.PlacesServiceStatus.OK) {
+          setNearByPlaces(results);
+          setError(false);
+        }
+      };
+      const request = {
+        keyword: searchTerm,
+        location: coordinates,
+        radius: '500',
+      };
+      const service = new mapsApi.places.PlacesService(mapInstance);
+      service && service.nearbySearch(request, callback);
+    }
+  }, [mapsApi, mapInstance, searchTerm, coordinates]);
+
+  return (
+    <div className="searchSection">
+      <form className="searchBar" onSubmit={(e) => handleSubmit(e)}>
```

Comment:
I’m pretty sure your `onSubmit` could be simplified to the following: `onSubmit={handleSubmit}`